config/management/src/secure_backend.rs

```diff
@@ -162,6 +163,55 @@ impl TryInto<Storage> for SecureBackend {
     }
 }
 
+macro_rules! secure_backend {
+    ($struct_name:ident, $field_name:ident, $struct_type:ty, $purpose:literal) => {
+        #[derive(Debug, StructOpt)]
+        pub struct $struct_name {
+            #[structopt(long,
+                help = concat!("Backend for ", $purpose),
+                long_help = concat!("Backend for ", $purpose, r#"
+
+Secure backends are represented as a semi-colon deliminted key value
+pair: "k0=v0;k1=v1;...".  The current supported formats are:
+    Vault: "backend=vault;server=URL;token=PATH_TO_TOKEN"
+        an optional namespace: "namespace=NAMESPACE"
+        an optional server certificate: "ca_certificate=PATH_TO_CERT"
+    GitHub: "backend=github;repository_owner=REPOSITORY_OWNER;repository=REPOSITORY;token=PATH_TO_TOKEN"
+        an optional namespace: "namespace=NAMESPACE"
+    InMemory: "backend=memory"
+    OnDisk: "backend=disk;path=LOCAL_PATH"
+                "#)
+            )]
+            pub $field_name: $struct_type,
+        }
+    }
+}
+
+secure_backend!(
```

Comment:
nit: I guess the only downside of using field names matching the struct name is that we have things like "self.validator_backend.validator_backend" in the code. Perhaps consider making all field names just be secure_backend so they're consistent?